### PR TITLE
CLOUDP-85812: use the atlas endpoint to get the public ip address of the user within quickstart

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-85812

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -247,7 +247,7 @@ func NewPrivateUnauth(c Config) (*Store, error) {
 	s.baseURL = atlas.CloudURL
 
 	if configURL := c.OpsManagerURL(); configURL != "" {
-		s.baseURL = s.apiPath(configURL)
+		s.baseURL = configURL
```
Comment (andreaangiolillo) [inline internal/store/store.go:250]: drive-by fix

Comment (andreaangiolillo) [inline internal/store/store.go:250]: drive-by fix

File: internal/net/net.go [hunk 2]
```diff
@@ -14,44 +14,20 @@
 package net
 
 import (
-	"context"
-	"io/ioutil"
-	"net/http"
-	"strings"
+	"github.com/mongodb/mongocli/internal/config"
```
Comment (gssbzn) [inline internal/net/net.go:17]: [q] should we keep this package? I feel we can be better off without and rely on our store package alone

File: internal/store/store.go (hunk 1 above)
Comment (gssbzn) [reply internal/store/store.go:250]: [q] are sure? this is used to create the owner and I'm pretty sure it was working as expected, remember that ops manager doesn't set up the `/api` segment as part of the base url

Comment (gssbzn) [reply internal/store/store.go:250]: also this is used for the default cluster version, I'll add cluster deployment tests to see if this works

Comment (gssbzn) [reply internal/store/store.go:250]: actually this may be right, I think this is only used for the default version and I think that one always uses teh default setting because of reasons

**Review by gssbzn — DISMISSED** [review]
LGTM left a question about maybe removing the net package, also I've bene wanting to refactor how we create clients, it's growing wild

File: internal/cli/atlas/quickstart/quick_start.go [hunk 3]
```diff
@@ -378,7 +377,7 @@ func (opts *Opts) askAccessListOptions() error {
 	}
 
 	message := ""
-	publicIP := net.IPAddress()
+	publicIP := store.IPAddress()
```
Comment (gssbzn) [inline internal/cli/atlas/quickstart/quick_start.go:380]: [q] any reason you can't do `opts.store.IPInfo()`?

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by andreaangiolillo — COMMENTED** [review]
@gssbzn I am not sure if your question is gone for a problem with github or if you deleted it. I will put my answer here anyway

I believe it wouldn’t work because I need to use `NewPrivate(config.Default())` to call `https://cloud-dev/api/private/ipInfo`. If I use directly `opts.store.IPInfo()`, I would call `https://cloud-dev/api/public/v1.0/api/private/ipInfo`.

**Review by andreaangiolillo — COMMENTED** [review]
I have just some questions to understand completely these changes. LGTM once my questions are answers. Thanks for all these changes!
